### ISSUE TITLE
Clarification for Memory Allocation Section

### ISSUE THREAD

**Issue by JamieBra** [Mukul1127/Minecraft-Performance-Flags-Benchmarks#20]
> sparkc gcmonitor will tell you if your allocation is too high (the pauses will be too long) or too low (frequent GC with a low memory warning in the notification).

For this part of the memory allocation section, it is unclear to me at least how to know if the pauses are too long and what a low memory warning looks like. For pauses being too long, is that something that you have to look at the milliseconds printed out by Spark to determine? How many milliseconds is a pause that is too long? For a low memory warning, where is it printed that the warning is caused by low memory?

If I have an example notification from running my game with 8192M and heavy shaders:
> Young Gen GC lasting 5 ms. (cause = G1 Evacuation Pause)
> 7.8 MB moved to G1 Old Gen
> 536.2 MB → 544.0 MB
> 5.0 MB freed from G1 Survivor Space
> 12.1 MB → 7.1 MB (41%)
> 1.8 GB freed from G1 Eden Space
> 1.8 GB → 0 bytes (100%)
Is 5 ms too long? At what point about is too long?

How frequent also is "frequent GC"? If i purposely set my memory extremely low while running heavy shaders (about 700M) I get spammed with these notifications:
> [15:26:36] [spark-worker-pool-1-thread-1/INFO]: [CHAT] [⚡] Old Gen GC lasting 257 ms. (cause = G1 Compaction Pause)\n  198.5 KB freed from CodeHeap 'profiled nmethods'\n    60.8 MB → 60.6 MB (0%)\n  64.0 MB freed from G1 Old Gen\n    651.4 MB → 587.4 MB (9%)\n  15.3 KB freed from CodeHeap 'non-profiled nmethods'\n    41.0 MB → 41.0 MB (0%)
> [15:26:36] [spark-worker-pool-1-thread-4/INFO]: [CHAT] [⚡] Young Gen GC lasting 2 ms. (cause = G1 Evacuation Pause)\n  64.0 MB moved to G1 Old Gen\n    587.4 MB → 651.4 MB\n  64.0 MB freed from G1 Eden Space\n    64.0 MB → 0 bytes (100%)
> [15:26:36] [spark-worker-pool-1-thread-2/INFO]: [CHAT] [⚡] Old Gen GC lasting 261 ms. (cause = G1 Compaction Pause)\n  208.1 KB freed from CodeHeap 'profiled nmethods'\n    60.8 MB → 60.6 MB (0%)\n  64.0 MB freed from G1 Old Gen\n    651.4 MB → 587.4 MB (9%)\n  75.1 KB freed from CodeHeap 'non-profiled nmethods'\n    41.3 MB → 41.2 MB (0%)
> [15:26:37] [spark-worker-pool-1-thread-1/INFO]: [CHAT] [⚡] Old Gen GC lasting 263 ms. (cause = G1 Compaction Pause)\n  745.9 KB freed from CodeHeap 'profiled nmethods'\n    60.7 MB → 59.9 MB (1%)\n  63.9 MB freed from G1 Old Gen\n    651.4 MB → 587.6 MB (9%)\n  139.8 KB freed from CodeHeap 'non-profiled nmethods'\n    41.6 MB → 41.5 MB (0%)
> [15:26:37] [spark-worker-pool-1-thread-3/INFO]: [CHAT] [⚡] Young Gen GC lasting 2 ms. (cause = G1 Evacuation Pause)\n  64.0 MB moved to G1 Old Gen\n    587.4 MB → 651.4 MB\n  64.0 MB freed from G1 Eden Space\n    64.0 MB → 0 bytes (100%)
> [15:26:37] [spark-worker-pool-1-thread-4/INFO]: [CHAT] [⚡] Young Gen GC lasting 2 ms. (cause = G1 Evacuation Pause)\n  64.0 MB moved to G1 Old Gen\n    587.6 MB → 651.6 MB\n  64.0 MB freed from G1 Eden Space\n    64.0 MB → 0 bytes (100%)
> [15:26:37] [spark-worker-pool-1-thread-2/INFO]: [CHAT] [⚡] Old Gen GC lasting 261 ms. (cause = G1 Compaction Pause)\n  320.3 KB freed from CodeHeap 'profiled nmethods'\n    60.1 MB → 59.8 MB (0%)\n  63.9 MB freed from G1 Old Gen\n    651.6 MB → 587.6 MB (9%)

Obviously at that speed it is too many, but there's also no clear indication if these are low memory notifications apart from that.

In a similar vein, it is not exactly clear what infrequent means here:
> Any old generation pauses are bad, and young generation G1GC collections should be infrequent, but short enough to be imperceptible.

I think some more clarification on these matters would be helpful for less tech-savvy people or people who are not familiar with garbage collection in Java and what the codes mean.

--

> Note: If you are using a minecraft launcher like Prism Launcher or ATLauncher, you shouldn't add memory arguments and instead control memory through the dedicated section in the launcher.

Also, I was testing with the Modrinth launcher which does have a memory allocation slider. However, when I inspect the command line arguments that it uses, only -Xmx8192M is added (if I set the slider to that many megabytes). Should I also add the -Xms8192M flag?

**Comment by Mukul1127** [comment]
I don't know exactly what this part means as this part is unedited from @brucethemoose's version and he never specified what he meant before he disappeared. Though probably just using a reasonable amount of ram will be fine for 99% of users. 

TLDR: Just use 4 - 8 GBs of ram and you'll probably be fine.

**Comment by Mukul1127** [comment]
I would assume that this part is more for micro optimizations so just use your own judgement or just ignore it.
